bin/ce

```diff
@@ -131,9 +142,11 @@ def restart_one_instance(as_group_name, instance, modified_groups):
         ShouldDecrementDesiredCapacity=not adjustment_required)
     wait_for_autoscale_state(instance, 'Standby')
     logger.info("Restarting service on {}".format(instance))
+    exec_remote(instance, ['sudo', 'docker', 'stop', 'nginx', 'unified'])
```

Comment:
The Stop occurs very late in the restart script, as https://github.com/mattgodbolt/compiler-explorer-image/blob/master/init/compiler-explorer.service doesn't have an actual Stop implementation - just a start.
And the (re)start only stops nginx just before starting https://github.com/mattgodbolt/compiler-explorer-image/blob/master/run_site.sh#L31. So the instance will be pretending to be /healthcheck Ok eventhough its files are being updated and replaced (which results in files returning 404's during when the loadbalancer points you to a certain instance)

So you could do it in the restart, but a Stop action would have to be written for compiler-explorer.service. That would probably be the best solution instead of this docker stop hack.

Reply:
I think so! Cool :) Maybe currently our issue is that we `restart` which takes long enough to stop and then restart that we miss the "falling edge" of going healthy->unhealthy. Then we end up thinking everything's OK and working and move on to the next node...but actually the restart hasn't even happened yet?

If it's OK with you let's make this a "stop" like you suggest (along with making a stop action)